src/components/CollectionProvider/CollectionProvider.tsx

```diff
@@ -1,28 +1,50 @@
 import * as React from 'react'
-import { Props } from './CollectionProvider.types'
+import equal from 'fast-deep-equal'
+import { DEFAULT_ITEMS_PAGE_SIZE, DEFAULT_ITEMS_PAGE, Props } from './CollectionProvider.types'
 
 export default class CollectionProvider extends React.PureComponent<Props> {
+  fetchCollectionItems(itemsPage: number | number[] = DEFAULT_ITEMS_PAGE) {
+    const { id, onFetchCollectionItems, onFetchCollectionItemsPages, itemsPageSize } = this.props
+    if (id) {
+      const pageSize = itemsPageSize || DEFAULT_ITEMS_PAGE_SIZE
+      Array.isArray(itemsPage) ? onFetchCollectionItemsPages(id, itemsPage, pageSize) : onFetchCollectionItems(id, itemsPage, pageSize)
```

Comment:
The different between the 2 actions here is that `onFetchCollectionItems` updates the state with the paginated data, and the one asking for multiple pages doesn't.  
Why? Because the use case for the TP Detail page is basically to fetch all the pages and don't modify the current paginated data (because the user is looking at a specific page at the moment of fetching all)

Reply:
Yeah, they looked pretty similar. I updated them so it's easier to follow, moving the logic that was the only thing different into the reducer. I think it's better now, let me know what you think :)